### PR TITLE
Code coverage reporting for C++ code

### PR DESCRIPTION
**Pull request by drewoldag**
Making modifications to the top level CMakeLists.txt and the github action `testing_and_coverage` in order to record and upload c++ code coverage numbers.

### CODE REVIEW

**Review by johannct — APPROVED** [review]
approving as we need to move forward, but standard build should not set coverage flags IIRC

File: CMakeLists.txt [hunk 1]
```diff
@@ -12,7 +12,7 @@ set(CMAKE_CXX_STANDARD_REQUIRED True)
 
 set(CMAKE_INSTALL_PREFIX "${CMAKE_SOURCE_DIR}")
 set(CMAKE_BINARY_DIR "${CMAKE_SOURCE_DIR}/bin")
-set(CMAKE_BUILD_TYPE "RelWithDebInfo")
+# set(CMAKE_BUILD_TYPE "RelWithDebInfo")
```
Comment (drewoldag) [inline CMakeLists.txt:15]: @johannct I noticed that this was hardcoded here. I've only commented it out because it was interfering with the conditional logic on line 69. I'm happy to revert this change if needed.

Comment (johannct) [reply CMakeLists.txt]: Can you expand on the overall take on conditional logic? RelWithDebInfo seems to be a standard for preformance builds but still with debug symbols. Perhaps we should lay down the different builds we need/want?

**Comment by johannct** [comment]
fantastic job on C++ coverage : https://app.codecov.io/gh/lephare-photoz/lephare/pull/94/blob/src/lib/SED.cpp#L36 !!! Thank so much

File: CMakeLists.txt (hunk 1 above)
Comment (drewoldag) [reply CMakeLists.txt]: Initially I wanted use `CMAKE_BUILD_TYPE="Debug"` be the trigger to include the coverage flags when building, but I'm happy to go back to the default that was here, and we'll look for an env var to conditionally include the coverage flags.
